core/src/epicli/data/common/validation/configuration/kubernetes-master.yml

```diff
@@ -1 +1,40 @@
-$ref: '#/definitions/unvalidated_specification'
\ No newline at end of file
+"$id": "#/specification"
+title: "K8s specification schema"
+description: "K8s specification schema"
+type: object
+properties:
+  advanced:
+    type: object
+    properties:
+      certificates:
+        type: object
+        properties:
+          location:
+            title: location schema
+            "$id": "#/properties/specification/properties/advanced/properties/certificates/properties/location"
+            description: |-
+              Where the certificates should be stored
+            type: string
+            default: /etc/kubernetes/pki
+            examples:
+            - /etc/kubernetes/pki
+          expiration_days:
+            title: expiration_days schema
+            "$id": "#/properties/specification/properties/advanced/properties/certificates/properties/expiration_days"
+            description: |-
+              Days to make certificate be valid for.
+              For more explanation about the limit navigate to
+              https://groups.google.com/g/mailing.openssl.users/c/3kK_f0ywCZQ.
+            type: integer
+            minimum: 1
+            maximum: 24855
+            default: 365
+          renew:
+            title: renew schema
+            "$id": "#/properties/specification/properties/advanced/properties/certificates/properties/renew"
+            description: |-
+              Whether to renew certificates or not
+            type: boolean
+            default: false
+            examples:
+            - false
```

Comment:
```suggestion
"$id": "#/specification"
title: kubernetes-master specification
type: object
properties:
  advanced:
    type: object
    properties:
      certificates:
        type: object
        properties:
          location:
            description: Where the certificates should be stored
            type: string
            default: /etc/kubernetes/pki
          expiration_days:
            description: >-
              Days to make certificate be valid for.
              For more explanation about the limit, navigate to
              https://groups.google.com/g/mailing.openssl.users/c/3kK_f0ywCZQ.
            type: integer
            minimum: 1
            maximum: 24855
            default: 365
          renew:
            description: Whether to renew certificates or not
            type: boolean
            default: false
```
What do you think about keeping this schema as compact as possible?
IMHO we may no need to define titles and examples (if `default` is defined which is also a kind of example 😉).

Having the above version I got all needed information to identify the incorrect setting:
```
14:05:06 INFO cli.engine.schema.SchemaValidator - Validating: configuration/kubernetes-master
14:05:06 ERROR cli.engine.schema.SchemaValidator - Failed validating: configuration/kubernetes-master
14:05:06 ERROR cli.engine.schema.SchemaValidator - 10 is not of type 'string'

Failed validating 'type' in schema['properties']['specification']['properties']['advanced']['properties']['certificates']['properties']['location']:
    {'default': '/etc/kubernetes/pki',
     'description': 'Where the certificates should be stored',
     'type': 'string'}

On instance['specification']['advanced']['certificates']['location']:
    10
14:05:06 INFO cli.engine.schema.SchemaValidator - Run done in 142ms
```

Reply:
Discussed with @to-bar, will be done later if necessary